api/logs/logs.go

```diff
@@ -6,26 +6,50 @@ import (
 	"strings"
 	"time"
 
+	durationUtil "github.com/flanksource/commons/duration"
 	"github.com/flanksource/kommons"
 )
 
 var GlobalBackends []SearchBackend
 
 type SearchConfig struct {
+	// Path is the path of this config file
+	Path     string          `yaml:"-"`
 	Backends []SearchBackend `yaml:"backends,omitempty"`
 }
 
 type SearchBackend struct {
-	Routes     []SearchRoute            `json:"routes,omitempty"`
-	Backend    SearchAPI                `json:"-"`
-	Kubernetes *KubernetesSearchBackend `json:"kubernetes,omitempty"`
-	Files      []FileSearchBackend      `json:"file,omitempty" yaml:"file,omitempty"`
+	Routes        []SearchRoute            `json:"routes,omitempty"`
+	Backend       SearchAPI                `json:"-"`
+	ElasticSearch *ElasticSearchBackend    `json:"elasticsearch,omitempty"`
+	OpenSearch    *OpenSearchBackend       `json:"opensearch,omitempty"`
+	Kubernetes    *KubernetesSearchBackend `json:"kubernetes,omitempty"`
+	Files         []FileSearchBackend      `json:"file,omitempty" yaml:"file,omitempty"`
 }
 
 type SearchRoute struct {
-	Type     string            `json:"type,omitempty"`
-	IdPrefix string            `json:"idPrefix,omitempty"`
-	Labels   map[string]string `json:"labels,omitempty"`
+	Type       string            `yaml:"type,omitempty"`
+	IdPrefix   string            `yaml:"idPrefix,omitempty"`
+	Labels     map[string]string `yaml:"labels,omitempty"`
+	IsAdditive bool              `yaml:"additive,omitempty"`
+}
+
+func (t *SearchRoute) Match(q *SearchParams) bool {
+	if t.Type != "" && t.Type != q.Type {
+		return false
+	}
+
+	if t.IdPrefix != "" && !strings.HasPrefix(q.Id, t.IdPrefix) {
+		return false
+	}
+
+	for k, v := range t.Labels {
+		if q.Labels[k] != v {
```

Comment:
I think should use the MatchItem format - i.e. split on "," - that would then enable us to support exsists `*`, not exists `!`